timechop/builders.py

```diff
@@ -164,46 +165,42 @@ def make_entity_date_table(
             )
         elif matrix_type == 'test':
             indices_query = self.get_all_valid_entity_date_combos(
-                as_of_times=as_of_times,
-                feature_table_names=feature_table_names
+                as_of_times=as_of_times
             )
         else:
             raise ValueError('Unknown matrix type passed: {}'.format(matrix_type))
 
-        table_name = '_'.join([matrix_uuid, 'tmp_entity_date'])
+        table_name = '_'.join(['tmp_entity_date', matrix_uuid])
         query = """
-            DROP TABLE IF EXISTS {features_schema_name}."{table_name}";
-            CREATE TABLE {features_schema_name}."{table_name}"
+            DROP TABLE IF EXISTS "{table_name}";
+            CREATE TEMPORARY TABLE "{table_name}"
             AS ({index_query})
         """.format(
-            features_schema_name=self.db_config['features_schema_name'],
             table_name=table_name,
             index_query=indices_query
         )
         self.engine.execute(query)
 
         return table_name
 
-    def get_all_valid_entity_date_combos(self, as_of_times, feature_table_names):
-        as_of_time_strings = [str(as_of_time) for as_of_time in as_of_times]     
-        query_list = []       
-        for index, table in enumerate(feature_table_names):
-            union = ''        
-            if index != 0:      
-                union = 'UNION'       
-            subquery = """ {u}        
-                SELECT DISTINCT entity_id, as_of_date     
-                FROM {schema_name}.{table_name}       
-                WHERE as_of_date IN (SELECT (UNNEST (ARRAY{dates}::timestamp[])))       
-            """.format(       
-                u = union,        
-                table_name = table,       
-                dates = as_of_time_strings,      
-                schema_name = self.db_config['features_schema_name']      
-            )     
-            query_list.append(subquery)
-        
-        return(''.join(query_list))
+    def get_all_valid_entity_date_combos(self, as_of_times):
+        as_of_time_strings = [str(as_of_time) for as_of_time in as_of_times]
+        query = """
+            SELECT entity_id, as_of_date
+            FROM (
+                SELECT DISTINCT entity_id
+                FROM {entities_table}
```

Comment:
Not all entities might exist at all times -- for instance, for San Jose, we only have MH permits for properties in certain time windows, so we should probably respect that the prediction set could be different given the as_of_time. One way to handle this might be to expect something like `entity_id`, `start_time` and `end_time` in the entities table and join into it where `as_of_date BETWEEN start_time AND end_time` to pick out the entities that exist/should be scored at a given point in time.